integrations/azure-devops/azure_devops/client/azure_devops_client.py

```diff
@@ -53,28 +54,35 @@ async def generate_members(self) -> AsyncGenerator[list[dict[str, Any]], None]:
                         member["__teamId"] = team["id"]
                     yield members
 
-    @cache_iterator_result("repositories")
-    async def generate_repositories(self) -> AsyncGenerator[list[dict[Any, Any]], None]:
+    @cache_iterator_result()
+    async def generate_repositories(
+        self, include_disabled_repositories: bool = True
+    ) -> AsyncGenerator[list[dict[Any, Any]], None]:
         async for projects in self.generate_projects():
             for project in projects:
                 repos_url = f"{self._organization_base_url}/{project['id']}/{API_URL_PREFIX}/git/repositories"
                 repositories = (await self.send_request("GET", repos_url)).json()[
                     "value"
                 ]
-                yield repositories
+                if include_disabled_repositories:
+                    yield repositories
+                else:
+                    yield [repo for repo in repositories if not repo.get("isDisabled")]
 
     async def generate_pull_requests(
         self, search_filters: Optional[dict[str, Any]] = None
     ) -> AsyncGenerator[list[dict[Any, Any]], None]:
-        async for repositories in self.generate_repositories():
+        async for repositories in self.generate_repositories(
+            include_disabled_repositories=False
+        ):
             for repository in repositories:
                 pull_requests_url = f"{self._organization_base_url}/{repository['project']['id']}/{API_URL_PREFIX}/git/repositories/{repository['id']}/pullrequests"
                 async for filtered_pull_requests in self._get_paginated_by_top_and_skip(
                     pull_requests_url, search_filters
                 ):
                     yield filtered_pull_requests
 
-    @cache_iterator_result("pipelines")
+    @cache_iterator_result()
```

Comment:
why do you need to cache the pipelines results? it can explode your cache and I don't see you re-use it in multiple as oppose to `generate_projects` / `generate_repositories`

Reply:
That's correct. will remove it